core/include/cubos/core/reflection/external/glm.hpp

```diff
@@ -0,0 +1,45 @@
+/// @file
+/// @brief Reflection declarations for external glm types.
+/// @ingroup core-reflection
+
+#pragma once
+
+#include <glm/fwd.hpp>
+
+#include <cubos/core/reflection/reflect.hpp>
+
+CUBOS_REFLECT_EXTERNAL_DECL(glm::quat);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dquat);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat2x3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat2x3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat2x4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat2x4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat3x2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat3x2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat3x4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat3x4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat4x2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat4x2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat4x3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat4x3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::mat4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dmat4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::vec2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::bvec2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dvec2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::ivec2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::uvec2);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::vec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::bvec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dvec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::ivec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::uvec3);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::vec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::dvec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::bvec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::ivec4);
+CUBOS_REFLECT_EXTERNAL_DECL(glm::uvec4);
```

Comment:
Some spacing would be appreciated between types of different 'groups' (mats vs vecs etc)